kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/json/JSONMarshaller.java

```diff
@@ -938,6 +945,16 @@ public Object deserializeTypedFromObject(JsonParser jp, DeserializationContext c
                 Thread.currentThread().setContextClassLoader(_baseType.getRawClass().getClassLoader());
                 if (classesSet.contains(_baseType.getRawClass()) && !jsonContext.get().isStripped()) {
 
+                    if (findDeserializerFirst) {
+                        String nextFieldName = jp.nextFieldName();
+//                        logger.info("nextFieldName : {}", nextFieldName);
+                        if (!baseTypeName().equals(nextFieldName)) {
+                            JsonDeserializer<Object> deser = _findDeserializer(ctxt, baseTypeName());
+                            Object value = deser.deserialize(jp, ctxt);
+                            return value;
+                        }
+                    }
+
```

Comment:
Please ignore this part because I set false for `findDeserializerFirst` in this PR.